### PR TITLE
Ldanzinger/viewshed

### PR DESCRIPTION
**Pull request by ldanzinger**
@michael-tims please review and merge

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+{
+  // Set a viewpoint
+  double x = 6.86088;
+  double y = 45.3604;
+  double z = 3582.55;
+  Point point(x, y, z, SpatialReference(4326));
+  double heading = 345;
+  double pitch = 70;
+  double roll = 0;
+  Camera camera(point, heading, pitch, roll);
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp]: Are these values stored in local variables for easier readability?

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+    }
+  });
+
+  connect(m_sceneView, &SceneQuickView::mouseReleased, this, [this](QMouseEvent)
+  {
+    m_calculating = false;
+  });
+}
+
+void ViewshedLocation::createViewshedTask(double x, double y)
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp]: It's not really a "task", so probably just call the method `createViewshed`

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+  }
+  else
+  {
+    if (m_obstructedColor == obstructedColor)
+      return;
+
+    m_obstructedColor = obstructedColor;
+  }
+
+  emit obstructedColorChanged();
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp]: The color methods should use the methods like they are static:
`Viewshed::visibleColor()`, `Viewshed::obstructedColor()`, etc...

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.qml [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                        onParameterValueChanged: viewshedSample.minDistance = parameterValue;
+                    }
+
+                    ViewshedSlider {
+                        titleText: qsTr("Max Distance (m)")
+                        parameterValue: viewshedSample.maxDistance
+                        minValue: 1
+                        maxValue: 2000
+                        onParameterValueChanged: viewshedSample.maxDistance = parameterValue;
+                    }
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.qml]: Optional... Min and Max distance would be nice as a RangeSlider.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.qml [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+                        maxValue: 179
+                        onParameterValueChanged: viewshedSample.horizontalAngle = parameterValue;
+                    }
+
+                    ViewshedSlider {
+                        titleText: qsTr("Heading")
+                        parameterValue: viewshedSample.heading
+                        minValue: 1
+                        maxValue: 359
+                        onParameterValueChanged: viewshedSample.heading = parameterValue;
```
Comment (michael-tims) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.qml]: Heading would be nice as a 360 dial, but not sure if one exists in Qt Quick Controls

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp (hunk 1 above)
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp]: Yes, we are trying to be more explicit in our samples. If you pass things into a function or constructor, and they take the same type for several arguments, it is ambiguous to a reader as to what the value is for, so they then would need to go to the doc to understand the code

Comment (michael-tims) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedLocation/ViewshedLocation.cpp]: Make sense 👍

**Pull request merged**